app/assets/javascripts/Components/Result/marks_panel.jsx

```diff
@@ -343,13 +344,13 @@ class RubricCriterionInput extends React.Component {
 
     return (
       <tr
-        data-level-index={i} onClick={() => this.handleChange(i)}
-        key={`${this.props.id}-${i}`}
+        data-level-index={levelMark} onClick={() => this.handleChange(levelMark)}
```

Comment:
You need to be careful about how you're calling `handleChange`. It expects an *index* and you're giving it a *mark*.

Similar issue with `data-level-index`. Try looking for where `data-level-index` is used to determine how to fix this mismatch. Note that JavaScript provides a way of accessing `data-*` attributes (https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes), so just doing a string search for 'data-level-index' won't necessarily find where it's being used.